src/circuit_optimizer/CircuitOptimizer.cpp

```diff
@@ -1434,9 +1435,10 @@ void CircuitOptimizer::collectBlocks(QuantumComputation& qc,
     bool canProcess = true;
     bool makesTooBig = false;
 
-    // check if the operation can be processed
     if (!op->isUnitary()) {
       canProcess = false;
+    } else if (onlyCollectCliffords && !op->isClifford()) {
+      continue;
```

Comment:
This is definitely not working as expected yet, which some improved tests would have shown. I recommend to improve the testing significantly to ensure that this is working as expected.
Non-Clifford operations need to finalize a block for a certain qubit. so that the next block comes _after_ the non-Clifford gate(s).